crates/store/src/state.rs

```diff
@@ -87,6 +96,43 @@ struct InnerState {
     account_tree: SimpleSmt<ACCOUNT_TREE_DEPTH>,
 }
 
+#[derive(Clone, Default, Debug)]
+pub struct NoteAuthenticationInfo {
+    pub block_proofs: Vec<BlockInclusionProof>,
+    pub note_proofs: BTreeMap<NoteId, NoteInclusionProof>,
+    pub chain_length: BlockNumber,
+}
```

Comment:
Not for this PR, but this should actually be in the domain objects of `miden-proto` crate (this is because this struct is used by both store and block producer). If we do that, we won't need to add `miden-store` as a dependency to `miden-block-producer` (I think).

Reply:
#455